Plane: QRTL: add interactive return altitude graph

IamPete1 opened the pull request:
Adds a interactive graph that changes with user inputs to show return altitude.

![image](https://user-images.githubusercontent.com/33176108/214967562-a8d56447-56fa-4b77-8148-5e4f8419caac.png)

Changing Q_RTL_ALT:
![image](https://user-images.githubusercontent.com/33176108/214967651-12b38cde-86fb-4e33-a51d-a293909b4747.png)

Someone who knows HTML better than I do could probably make it neater. Its possible there are more integrated ways to do this, I have just embeded HTML into the page and the script into the HTML. 

I do think this could be quite a powerful way to show things. It also provides a graph that could be copied into a operations manual.

IamPete1 commented:
I should probably add that the feature that this shows is only in master ATM.

Hwurzburg commented:
looks good! I would put this on the future-wiki-changes.rst under the Plane site....I can help if you wish...also, is the gitignore change part of this, I think not, right?might tweak the size of the legend lines....

could we do one of these for the hybrid RTL also??? pretty please....

IamPete1 commented:
@Hwurzburg yeah, gitignore changes are unrelated, I just had a bunch of new stuff show up after a build.

Yeah I can help with more, would be good to see if there is a way to make this more integrated, just including a bunch of HTML is not ideal, having said that this does seem to work..... I just wanted to see if it was possible to add interactive elements.

IamPete1 commented:
I did a little reading, I think we could make it so you could insert the canvas like a image and then insert each param with user input box. You would then just need to write the custom javascript stuff to populate it. However to do that we would need to write a custom sphinx extension with I have never done, so it might not be worth it.....,

IamPete1 commented:
After a good deal of messing about I have improved the formatting of the labels and removed most of the boiler plate html that was embeded.

Now looks like this:
![image](https://user-images.githubusercontent.com/33176108/215005921-3b2cd841-452f-4572-90dc-a18ea7998c5d.png)

Formatting does get messed up with a narrow window doing wrapping.

![image](https://user-images.githubusercontent.com/33176108/215005981-329541f6-ed4a-4520-b747-39675242da97.png)

Does come good again if you go very narrow....
![image](https://user-images.githubusercontent.com/33176108/215006027-ede34ce2-5a87-46ec-bf93-cf32601d4f60.png)

I have no idea how to fix that..... I think its better than starting out with one on each line.

IamPete1 commented:
Getting carried away now....

I have added a return path, you can click the graph and it show the path home. 

![image](https://user-images.githubusercontent.com/33176108/215012207-63c02d27-e0f8-4d42-9004-666de360b801.png)

![image](https://user-images.githubusercontent.com/33176108/215012267-edebfca2-54fd-4420-a21b-2e66d32d222e.png)

![image](https://user-images.githubusercontent.com/33176108/215012279-8450b430-fafa-42f7-8c26-e71758437cdc.png)

IamPete1 commented:
Moved to future section. Got the legend looking better and fixed the axis lables.

![image](https://user-images.githubusercontent.com/33176108/215189386-2ff9fad2-22fb-4e70-b2f5-22e2ed3804da.png)

If you click it adds the return path. Sometimes with FW sometimes not:

![image](https://user-images.githubusercontent.com/33176108/215189446-9f31f481-2b42-4f9e-a697-a2711f4d9c91.png)

![image](https://user-images.githubusercontent.com/33176108/215189531-74a620f8-ecb3-4c2c-a2be-16079c0792dd.png)

![image](https://user-images.githubusercontent.com/33176108/215189560-053dff44-e816-4e46-90a9-366f7e73c808.png)

Hwurzburg commented:
@IamPete1 I think the second graph is wrong....long FW return shoots for Q_RTL_ALT at 1.5X maxrad, then airbrakes in POS1, moves to POS2, then decends, correct?

IamPete1 commented:
@Hwurzburg I have removed the sloped FW return, so now it looks like this:
![image](https://user-images.githubusercontent.com/33176108/219871648-f13e1ed2-cf86-4cb2-b743-7626c454f144.png)

There is other stuff that will happen at larger radius's. It will climb up to RTL alt and then descend down again. But that all happens between 2 and 20 times the radius. This graph stops at 2 times the radius. I don't think we can show that on this graph without making the bit close to home very tiny.

We could show that on a second graph of course. Or make the max display radius display configurable but default to as it is now.

Hwurzburg commented:
@IamPete1 lets get rid of anyting FW because what is shown is not accurate....lets just talk to what happens  beyond 2X ...ie transition to FW and do a hybrid RTL

IamPete1 commented:
@Hwurzburg FW is correct now more or less, once you start the transition it will hold altitude. So there might be a little wobble at the start but then it will maintain altitude.

Hwurzburg commented:
Sorry, but that is not correct....here is what your current graph in this PR shows if you switch to QRTL from VTOL at alt above Q_RTL_ALT and beyond 1.5X:
![Screenshot from 2023-02-21 07-39-13](https://user-images.githubusercontent.com/6076534/220361752-178c7ea3-7f60-458b-9eaa-8d6fe9a7b6c6.png)
here is what it actually does in SITL (as I would expect):
![qrtl_fs_return](https://user-images.githubusercontent.com/6076534/220362008-571c6e8b-bba7-4c07-a46a-27e2b536cff3.png)

lets just remove FW anything from the graph and make the right margin just under 1.5X from home...it will be less confusing

Hwurzburg reviewed (changes requested):
lets remove the FW line

Hwurzburg commented:
@IamPete1 bump....if we get rid of the FW return line we could add this

Hwurzburg commented:
@IamPete1 closing this since its been forever and never worked on...feel free to reopen
